test-data/unit/check-isinstance.test

```diff
@@ -2207,3 +2207,23 @@ def bar(x: Union[List[str], List[int], None]) -> None:
     assert isinstance(x, list)
     reveal_type(x)                  # E: Revealed type is 'Union[builtins.list[builtins.str], builtins.list[builtins.int]]'
 [builtins fixtures/isinstancelist.pyi]
+
+[case testIsInstanceWithStarExpression]
+var = 'some string'
```

Comment:
This should be initially a union or a superclass (like `object`). So that below you can see some effect of the `isinstance()` check.

Reply:
Done.